static/js/components/request.vue

```diff
@@ -128,8 +128,8 @@ export default {
       }
     }, 300),
     instrumentToDataType: function(value){
-      if(!_.isEmpty(this.available_instruments) && value) {
-        return this.available_instruments[value].type;
```

Comment:
Make sure this first part will never be called (after this.available_instruments is populated)

Reply:
It's only called once. I added a log message to this method, and it was only logged once, even after this.available_instruments is populated.